File: Java/src/test/java/SudokuValidatorTest.java
import io.github.joaorbrandao.SudokuValidator;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SudokuValidatorTest {
    @Test
    void itShouldSayItIsValid() {
        List<List<Character>> given = List.of(
            List.of('5','3','.','.','7','.','.','.','.')
            ,List.of('6','.','.','1','9','5','.','.','.')
            ,List.of('.','9','8','.','.','.','.','6','.')
            ,List.of('8','.','.','.','6','.','.','.','3')
            ,List.of('4','.','.','8','.','3','.','.','1')
            ,List.of('7','.','.','.','2','.','.','.','6')
            ,List.of('.','6','.','.','.','.','2','8','.')
            ,List.of('.','.','.','4','1','9','.','.','5')
            ,List.of('.','.','.','.','8','.','.','7','9')
        );

        var validator = new SudokuValidator(given);
        var result = validator.evaluate();
        assertEquals("Valid", result);
    }
}

File: Kotlin/src/test/kotlin/SudokuValidatorTest.kt
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class SudokuValidatorTest {
    @Test
    fun `it should say it is valid`() {
        val given = listOf(
            listOf('5','3','.','.','7','.','.','.','.')
                ,listOf('6','.','.','1','9','5','.','.','.')
                ,listOf('.','9','8','.','.','.','.','6','.')
                ,listOf('8','.','.','.','6','.','.','.','3')
                ,listOf('4','.','.','8','.','3','.','.','1')
                ,listOf('7','.','.','.','2','.','.','.','6')
                ,listOf('.','6','.','.','.','.','2','8','.')
                ,listOf('.','.','.','4','1','9','.','.','5')
                ,listOf('.','.','.','.','8','.','.','7','9')
        )

        val validator = SudokuValidator(given)
        val result = validator.evaluate()
        assertEquals("Valid", result)
    }
}